### PR TITLE
Release 0.15

### PR DESCRIPTION
**Pull request by rnicoll**

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -5,36 +5,38 @@ version = '0.15-SNAPSHOT'
 archivesBaseName = 'libdohj-core'
 
 dependencies {
-    compile 'org.bitcoinj:bitcoinj-core:0.15.9'
-    compile 'com.madgag.spongycastle:core:1.58.0.0'
+    implementation 'org.bouncycastle:bcprov-jdk15to18:1.68'
```
Comment (langerhans) [inline core/build.gradle]: That completely breaks on Android. Using BC is a huge pain on Android due to conflicts with the sytsem supplied version.

File: core/build.gradle [hunk 2]
```diff
@@ -5,36 +5,38 @@ version = '0.15-SNAPSHOT'
...
+    implementation 'org.bitcoinj:bitcoinj-core:0.15.10'
```
Comment (langerhans) [inline core/build.gradle]: Change to 
```suggestion
    api 'org.bitcoinj:bitcoinj-core:0.15.10'
```

This requires to also change java to java-library on line 1

File: core/build.gradle [hunk 3]
```diff
@@ -5,36 +5,38 @@ version = '0.15-SNAPSHOT'
...
-    implementation 'com.squareup.okhttp3:okhttp:3.9.1'
-    implementation 'org.slf4j:slf4j-api:1.7.25'
+    implementation 'com.google.protobuf:protobuf-java:3.13.0'
+    implementation 'com.squareup.okhttp3:okhttp:3.12.12'
+    implementation 'org.slf4j:slf4j-api:1.7.30'
     implementation 'net.jcip:jcip-annotations:1.0'
-    compileOnly 'org.fusesource.leveldbjni:leveldbjni-all:1.8'
-    testImplementation 'junit:junit:4.12'
-    testImplementation 'org.easymock:easymock:3.2'
+    implementation 'org.fusesource.leveldbjni:leveldbjni-all:1.8'
```
Comment (langerhans) [inline core/build.gradle]: Not sure about that change. compileOnly is s till a thing even in newer Gradle versions.

**Review by langerhans — CHANGES_REQUESTED** [review]
Built with the mentioned changes on Android and works. Without the changes it doesn't compile, respectively crashes when accessing a BC API.

File: core/build.gradle (hunk 1 above)
Comment (rnicoll) [reply core/build.gradle]: Does it still break? bitcoinj moved over to Bouncycastle ( https://github.com/bitcoinj/bitcoinj/blob/master/core/build.gradle#L11) so I presumed Android had unscrewed things.

If not, I'd have to make a new spongycastle as 1.58 was the last one.

Comment (rnicoll) [reply core/build.gradle]: https://github.com/rtyley/spongycastle/issues/34 is relevant

File: core/build.gradle (hunk 2 above)
Comment (rnicoll) [reply core/build.gradle]: Done

File: core/build.gradle (hunk 3 above)
Comment (rnicoll) [reply core/build.gradle]: Reverted

**Review by langerhans — APPROVED** [review]

**Pull request merged**